be/src/storage/vectorized/reader.cpp

```diff
@@ -217,8 +215,8 @@ Status Reader::_init_delete_predicates(const ReaderParams& params, DeletePredica
             TCondition cond;
             if (!DeleteHandler::parse_condition(pred_pb.sub_predicates(i), &cond)) {
                 LOG(WARNING) << "invalid delete condition: " << pred_pb.sub_predicates(i) << "]";
-                st = Status::InternalError("invalid delete condition string");
-                break;
+                _tablet->release_header_lock();
+                return Status::InternalError("invalid delete condition string");
```

Comment:
Keep one function only one return statement may be better.

Reply:
The purpose is exactly early return once error occurred to avoid the subsequent procedure.